src/EnergyPlus/api/datatransfer.cc

```diff
@@ -160,7 +160,7 @@ void requestVariable(EnergyPlusState state, const char *type, const char *key)
     // allow specifying a request for an output variable, so that E+ does not have to keep all of them in memory
     // should be called before energyplus is run!
     // note that the variable request array is cleared during clear_state, so if you run multiple E+ runs, these must be requested again each time.
-    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
```

Comment:
EnergyPlus::EnergyPlusData *thisState ?

Reply:
@rraustad changing it to explicitly declare the type instead of auto would be perfectly fine too.  In (most?) IDEs, if you do a reinterpret_cast or one of the other casting types, the IDEs tell you that you should use auto to avoid writing the type in 2 spots in the same line.  I have no preference here whatsoever, but this is perfectly fine.

Reply:
`auto` can differentiate between values and pointers because they have different signatures, but it cannot differentiate between values and references because references are pointers but have the type signatures of values. Because we use references so extensively, we need to use `auto &` which means that just plain `auto` looks like a value. And because we are limiting the number of idioms in which value `auto` is used, I am adding this `*` to tell my brain and my `grep` scripts that this is a perfectly fine use of `auto`.